### PR TITLE
Stuff2

### PR DESCRIPTION
**Pull request by MissAllSunday**
- Remove storing the entire file in $_SESSION, it now only stores the ID.
- Adds a generic (blue) progress bar and removes unneeded css for the other progress bars.
- Adds the remaining fixes added to 2.0.12

### CODE REVIEW

File: Sources/Subs-Attachments.php [hunk 1]
```diff
@@ -1002,6 +988,52 @@ function parseAttachBBC($attachID = 0)
...
+	global $smcFunc, $modSettings;
+
+	if (empty($attachIDs))
+		return array();
+
+	$return = array();
+
+	$request = $smcFunc['db_query']('', '
+		SELECT a.id_attach, a.id_msg, a.id_member, a.size, a.mime_type, a.id_folder, a.filename' . (empty($modSettings['attachmentShowImages']) || empty($modSettings['attachmentThumbnails']) ? '' : ',
+				IFNULL(thumb.id_attach, 0) AS id_thumb, thumb.width AS thumb_width, thumb.height AS thumb_height') . '
```
Comment (albertlast) [inline Sources/Subs-Attachments.php]: IFNULL support was dropped.

Comment (live627) [inline Sources/Subs-Attachments.php]: https://github.com/SimpleMachines/SMF2.1/pull/3277

Comment (albertlast) [inline Sources/Subs-Attachments.php]: nope this one drop the support: https://github.com/SimpleMachines/SMF2.1/pull/3278

**Comment by Underdog-01** [comment]
.. just thought I'd mention this on an open commit so it goes noticed.
Some changes require an update to the SMF 2.1 minimal requirements reference which should be edited more frequently to refelct these changes.
out of date ref. http://wiki.simplemachines.org/smf/SMF2.1:Features#Minimum_requirements
COALESCE requires MySQL 5.6
Imo a minimal requirements reference should be included on the GitHub repository resource to make things easier. That way the SMF wiki page can be updated for releases posted on the SMF website itself.

**Comment by albertlast** [comment]
Like I write in the pr: coalesce exists with mysql 5.0 http://dev.mysql.com/doc/refman/5.0/en/comparison-operators.html
pg min version is in documentation outdated should be 8.3 (full text search)

File: Sources/Subs-Attachments.php (hunk 1 above)
Comment (live627) [inline Sources/Subs-Attachments.php]: I posted for reference -- should've been more clear, sorry.

**Comment by live627** [comment]
Both your PRs have conflicts now that I merged one.

I feel like such a burden. I'm sorry...

**Pull request merged**